rest/src/main/java/fi/otavanopisto/pyramus/rest/PersonRESTService.java

```diff
@@ -329,7 +331,19 @@ public Response updateCredentials(@PathParam("ID") Long id, UserCredentials user
                 userIdentificationDAO.updateExternalId(userIdentification, externalId);
               } else {
                 // Check that old password matches
+                User userByName = internalAuthenticationProvider.getUserByName(newUserName);
+                boolean usernameExists;
+                
+                if (userByName != null) {
```

Comment:
If the username belongs to the user who is changing their credentials then that's perfectly fine. With this implementation, the user is forced to also change their username whenever they want to change their password :)